designs/DotNetSignCommands.md

```diff
@@ -0,0 +1,140 @@
+
+## Non-Goals
+
+Adding additional signing-related commands, or significantly
+changing/improving the behavior of the commands as currently designed.
+
+## Solution
+
+The following commands will be implemented in the `dotnet.exe` CLI.
+
+Options marked with `*` indicate a notable change in behavior/naming.
+
+### `dotnet nuget sign`
+
+```
+Signs a NuGet package at <package-path> with the specified certificate.
+
+USAGE:
+    dotnet nuget sign [OPTIONS] <package-path>
+
+OPTIONS:
+        --output-directory              Directory where the signed package should be saved. By default, the original package is overwritten by the signed package.
+        --certificate-path              File path to the certificate to be used while signing the package.
+        --certificate-store-name        Name of the X.509 certificate store. (...)
+        --certificate-store-location    Name of the X.509 certificate store to use to search for the certificate.
+        --certificate-subject-name      Subject name of the certificate used to search a local certificate store for the certificate.
+        --certificate-fingerprint       SHA-1 fingerprint of the certificate used to search a local certificate store for the certificate.
+        --certificate-password          Password for the certificate, if needed.
+        --hash-algorithm                Hash algorithm to be used to sign the package. Defaults to SHA256.
+        --timestamper                   URL to an RFC 3161 timestamping server.
+        --timestamp-hash-algorithm      Hash algorithm to be used by the RFC 3161 timestamping server. Defaults to SHA256.
+        --overwrite                     Switch to indicate if the current signature should be overwritten.
+    -h, --help                          Prints help information.
+    -v, --verbosity* <LEVEL>            Set the MSBuild verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+        --interactive*                  Allows command to stop and wait for user input or action.
+
+EXAMPLES:
+
+    dotnet nuget sign MyPackage.nupkg --timestamper https://foo.bar
+
+    dotnet nuget sign ../MyPackage.nupkg --timestamper https://foo.bar --output-directory ../signed
+```
+
+Notable deviations from `nuget.exe` version:
+
+- No `-ConfigFile` equivalent. `dotnet.exe` doesn't seem to do this at all.
+- No `-ForceEnglishOutput` equivalent. `dotnet.exe` doesn't seem to do this at all.
+- Use MSBuild verbosity levels for `--verbosity` instead of NuGet levels.
+- Use `--interactive` instead of `--non-interactive`, because that's how `dotnet.exe` does it.
+
+### `dotnet nuget verify`
+
+```
+Verifies a signed NuGet package at <package-path>.
+
+USAGE:
+    dotnet nuget verify [OPTIONS] <package-path>...*
+
+OPTIONS:
+        --all-verifications*                    Specifies that all verifications possible should be performed on the package(s). By default, only signatures are verified.
```

Comment:
Should we simplify to `--All`? 

`dotnet nuget verify --all-verifications` is a few extra keystrokes if we believe "verifications" can be inferred by `--All` or `--all`.